### PR TITLE
minimum python -> 3.6 (#829)

### LINKED ISSUE #829: Remove Python 3.5 support
As it has reached [End-of-Life](https://www.python.org/dev/peps/pep-0478/#release-schedule), we can safely remove support and update docs.

Noting hear as there were some challenges that Python 3.5 presented specifically when building lightweight conda-based images for development, testing, and deployments.

### PR DESCRIPTION
**Pull request by ixcat**
fix #829 

Bump minimum required python from 3.5 to 3.6, remove 3.5 / add 3.9 to
unit test actions.

Currently, this is a configuration-only change, removal of OrderedDict,
etc. or other >3.5.x feature migration not included.

### CODE REVIEW

**Comment by ixcat** [comment]
@guzman-raphael - since this touched the GH actions somewhat blindly, would be good to get your signoff / adjustments.

**Comment by guzman-raphael** [comment]
@ixcat The issue here is b/c we have not yet added proper support for python3.9 in our images and also have not migrated the images used in tests here to utilize the new, optimized image hierarchy. The right thing to do would be to pull from https://github.com/datajoint/djtest-docker and add `3.9` to its build/push process.

**Comment by ixcat** [comment]
makes sense - is that your preferred course of action, or to back out the 3.9 additions here and just leave the 3.5 removals?

**Comment by guzman-raphael** [comment]
@ixcat I think for now, simply removing Python3.5 is fine here. We do need to add Python3.9 to the process but it comes with some time for development to propagate, verify things. In any case, it should be low risk as I am not aware of any feature we are leveraging that is currently throwing a deprecation warning.

**Comment by ixcat** [comment]
think this is gtg @guzman-raphael

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,8 +1,11 @@
 ## Release notes
 
+### Current
+=======
```
Comment (guzman-raphael) [inline CHANGELOG.md:4]:
```suggestion
```
Not sure if you intentionally meant for this to go in but the `0.13.0 -- TBD` block is serving as the 'current' placeholder.

**Review by guzman-raphael — CHANGES_REQUESTED** [review]
Thanks for the update, @ixcat. We'll also need to coordinate a new PR to our [conda forge feedstock](https://github.com/conda-forge/datajoint-feedstock) as well for this to be fully applied across distributions. We can handle that change though once we are closer to `0.13.0` release though.

File: CHANGELOG.md (hunk 1 above)
Comment (ixcat) [reply CHANGELOG.md]: yes, this is a side-effect from 'develop against master', wasn't sure of the best approach. 

updated/adjusted/conflict resolved. would be good to figure out the best approach to auto-prevent conflicts here if possible, though these are quick to resolve anyway.

**Review by guzman-raphael — CHANGES_REQUESTED** [review]
@ixcat Since I merged #859, this now also has a conflict. Could you resolve this? This is expected when you modify the same file across different PR's (as why it is only the `CHANGELOG.md`).

Speaking of which, please go ahead and update `docs-parts/intro/Releases_lang1.rst` document as well.

**Comment by ixcat** [comment]
@guzman-raphael resolved merge conflicts

**Comment by ixcat** [comment]
sorry, @guzman-raphael docs-parts/intro/Releases_lang1.rst also updated now.

**Review by guzman-raphael — APPROVED** [review]

**Pull request merged**